api/rosetta/data_transaction_integration_test.go

```diff
@@ -0,0 +1,554 @@
+			wantRosettaErrorDescription: fmt.Sprintf("transaction identifier: hash field has wrong length (have: %d, want: %d)", len(trimmedTxHash), validBlockHashLen),
+		},
+		{
+			name: "invalid transaction id",
+			request: rosetta.TransactionRequest{
+				NetworkID: defaultNetwork(),
+				BlockID:   testBlock,
+				TransactionID: identifier.Transaction{
+					Hash: invalidTxHash,
+				},
+			},
+
+			wantStatusCode:              http.StatusUnprocessableEntity,
+			wantRosettaError:            configuration.ErrorInvalidTransaction,
+			wantRosettaErrorDescription: "transaction hash is not a valid hex-encoded string",
+			wantRosettaErrorDetails:     map[string]interface{}{"hash": invalidTxHash},
+		},
+		// TODO: add - transaction that has no events/transfers
+		// TODO: add - transaction that does not exist in a block
+		// 	=> https://github.com/optakt/flow-dps/issues/195
+	}
+
+	for _, test := range tests {
+
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+
+			t.Parallel()
+
+			// prepare request payload
+			enc, err := json.Marshal(test.request)
+			require.NoError(t, err)
+
+			// create the request
+			req := httptest.NewRequest(http.MethodPost, "/block/transaction", bytes.NewReader(enc))
+			req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+			rec := httptest.NewRecorder()
+
+			ctx := echo.New().NewContext(req, rec)
+
+			// execute the request
+			err = api.Transaction(ctx)
+			assert.Error(t, err)
+
+			echoErr, ok := err.(*echo.HTTPError)
+			require.True(t, ok)
+
+			// verify HTTP status code
+			assert.Equal(t, test.wantStatusCode, echoErr.Code)
+
+			gotErr, ok := echoErr.Message.(rosetta.Error)
+			require.True(t, ok)
+
+			assert.Equal(t, test.wantRosettaError, gotErr.ErrorDefinition)
+			assert.Equal(t, test.wantRosettaErrorDescription, gotErr.Description)
+			assert.Equal(t, test.wantRosettaErrorDetails, gotErr.Details)
+		})
+	}
+}
+
+func TestMalformedTransactionRequest(t *testing.T) {
+
+	db := setupDB(t)
+	api := setupAPI(t, db)
+
+	const (
+		// network field is an integer instead of a string
+		wrongFieldType = `
+		{ 
+			"network_identifier": { 
+				"blockchain": "flow", 
+				"network": 99
+			}
+		}`
+
+		unclosedBracket = `
+		{
+			"network_identifier" : {
+				"blockchain": "flow",
+				"network": "flow-testnet"
+			},
+			"block_identifier": {
+				"index": 106,
+				"hash": "1f269f0f45cd2e368e82902d96247113b74da86f6205adf1fd8cf2365418d275"
+			},
+			"transaction_identifier": {
+				"hash": "071e5810f1c8c934aec260f7847400af8f77607ed27ecc02668d7bb2c287c683"
+			}`
+
+		validJSON = `
+		{
+			"network_identifier" : {
+				"blockchain": "flow",
+				"network": "flow-testnet"
+			},
+			"block_identifier": {
+				"index": 106,
+				"hash": "1f269f0f45cd2e368e82902d96247113b74da86f6205adf1fd8cf2365418d275"
+			},
+			"transaction_identifier": {
+				"hash": "071e5810f1c8c934aec260f7847400af8f77607ed27ecc02668d7bb2c287c683"
+			}
+		}`
+	)
+
+	tests := []struct {
+		name     string
```

Comment:
I would suggest using `description` or `desc` over `name`, as `name` can often conflict with the test vector's input variables

Reply:
I don't mind renaming it to `description` or `desc`.

But since we're running the tests as `t.Run()` and the `Run()` function declaration states:

```go
// Run runs f as a subtest of t called name. It runs f in a separate goroutine
// and blocks until f returns or calls t.Parallel to become a parallel test.
// Run reports whether f succeeded (or at least did not fail before calling t.Parallel).
//
// Run may be called simultaneously from multiple goroutines, but all such calls
// must return before the outer test function for t returns.
func (t *T) Run(name string, f func(t *T)) bool {
```

do you think that `name` perhaps fits better?

Reply:
@Ullaakut, you just called it _test vector_ :D

I agree with Čeki here, since that's what the standard library uses.

Reply:
Alright then let's use `name` everywhere from now on to stay consistent 👍 The difference is minimal, and I guess if we have a `name` input we can always prefix it with `input` or smth like that.